docs/command_reference.md

```diff
@@ -0,0 +1,279 @@
+a private key has been compromised.
+
+When using the Notary CLI client you need to specify where is Notary server
+you want to communicate with with the `-s` flag, and where to store the private keys and cache for
+the CLI client with the `-d` flag.  There are also fields in the [client configuration](/reference/client-config.md) to set these variables.
+
+```bash
+# Create an alias to always have the notary client talking to the right notary server
+$ alias notary="notary -s <notary_server_url> -d <notary_cache_directory>
+```
+
+When working Docker Content Trust, it is important to connect to Docker Hub's Notary server located at `https://notary.docker.io`, and specify notary's metadata cache as `~/.docker/trust`.
+
+## Initializing a trusted collection
+
+Notary can initialize a trusted collection with the `notary init` command:
+```bash
+$ notary init <GUN>
+```
+
+This command will generate targets and snapshot keys locally for the trusted collection, and try to locate a root key to use in the specified metadata cache (from `-d` or the config).
+If notary cannot find a root key, it will generate one.  For all keys, notary will also prompt for a passphrase to encrypt the private key material at rest.
+
+If you'd like to initialize your trusted collection with a specific root key, there is a flag:
+```bash
+$ notary init <GUN> --rootkey <key_file>
+```
+
+Note that you will have to run a publish after this command, as there are staged changes to initialize the trusted collection that have not been pushed to a notary server.
+```bash
+$ notary publish <GUN>
+```
+
+## Manage staged changes
+
+The Notary CLI client stages changes before publishing them to the server.
+These chagnes are staged locally in files such that the client can request the latest updates from the notary server, attempt to rebase the staged changes against new updates, and then finally publish if the changes are still valid.
+
+You can manage the changes that are staged by running:
+
+```bash
+# Check what changes are staged
+$ notary status <GUN>
+
+# Unstage a specific change
+$ notary status <GUN> --unstage 0
+
+# Alternatively, unstage all changes
+$ notary status <GUN> --reset
+```
+
+Note that `<GUN>` can take on arbitrary values, but when working with Docker Content Trust they are structured like `<url>/<account>/<repository>`.  For example `docker.io/library/alpine` for the [Docker Library Alpine image](https://hub.docker.com/r/library/alpine).
+
+When you're ready to publish your changes to the Notary server, run:
+
+```bash
+$ notary publish <GUN>
+```
+
+## Auto-publish changes
+
+Instead of manually running `notary publish` after each command, you can use the `-p` flag to auto-publish the changes from that command.
+For example:
+
+```bash
+$ notary init -p <GUN>
+```
+
+
+## Adding and removing trust data
+
+Users can sign content into a notary trusted collection by running:
+```bash
+$ notary add -p <GUN> <target_name> <target_file>
```

Comment:
I wanted to keep `-p` in commands after introducing it in the auto-publish section so that all of the commands will complete the full publish cycle.  I'll make a note in the previous section about how it'll be included going forward for commands that allow it (and that it is optional)